plugins/main/public/components/common/search-bar/use-search-bar.ts

```diff
@@ -48,58 +33,61 @@ type tUserSearchBarResponse = {
  * @returns
  */
 const useSearchBarConfiguration = (
-  props?: tUseSearchBarProps,
+  props: tUseSearchBarProps,
 ): tUserSearchBarResponse => {
+  const { indexPattern, filters: defaultFilters, setFilters } = props;
   // dependencies
-  const SESSION_STORAGE_FILTERS_NAME = 'wazuh_persistent_searchbar_filters';
-  const SESSION_STORAGE_PREV_FILTER_NAME = 'wazuh_persistent_current_filter';
-  const filterManager = useFilterManager().filterManager as FilterManager;
-  const filters = props?.filters ? props.filters : filterManager.getFilters();
+  const filters = defaultFilters ? defaultFilters : [];
   const [query, setQuery] = props?.query
     ? useState(props?.query)
     : useQueryManager();
   const { timeFilter, timeHistory, setTimeFilter } = useTimeFilter();
   // states
-  const [isLoading, setIsLoading] = useState(false);
-  const [indexPatternSelected, setIndexPatternSelected] =
-    useState<IIndexPattern>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [indexPatternSelected, setIndexPatternSelected] = useState<IndexPattern>(indexPattern);
+  const TIMEOUT_MILISECONDS = 100;
+
+  const hideRemoveFilter = (retry: number = 0) => {
+    let elements = document.querySelectorAll('.wz-search-bar .globalFilterItem');
+    if ((!elements || !filters.length) && retry < 10) {
+      // the setTimeout is used to wait for the DOM elements to be rendered
+      setTimeout(() => {
+        // this is a workaround to hide the close button on fixed filters via vanilla js
+        hideRemoveFilter(++retry);
+      }, TIMEOUT_MILISECONDS);
+
+    }else{
+      hideCloseButtonOnFixedFilters(filters, elements);
+    }
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      // this is a workaround to hide the close button on fixed filters via vanilla js
+      hideRemoveFilter();
+    }, TIMEOUT_MILISECONDS);
+  }, [filters]);
```

Comment:
Since the functionality of `hideRemoveFilter` depends on the DOM being updated to interact with its elements. It may be more appropriate to use `useLayoutEffect` instead of `useEffect`.
_Note: this will not improve the flickering that occurs with the x of the filter, since it does not depend solely on the rendering._